### PR TITLE
Export to RADAR

### PR DESCRIPTION
**Pull request by jochenklar**
Add export to RADAR functionality incl. a Metadata model to store RADAR/DataCite metadata for a collection.

### CODE REVIEW

**Comment by jochenklar** [comment]
Most of the functionality is implemented:

* Add Metadata model and API to create a RADAR/Datacite object (as JSON) for each Collection
* Add front end to edit metadata. Title and first creator is copied from existing data.
* Add API, Job, Config and Lib to create a Dataset in RADAR and upload the exported collection.
* Store radarId and radarUrl in Metadata after creation in RADAR.

Still missing:

* Update of previously created Dataset in RADAR.

**Comment by jochenklar** [comment]
The title in RADAR is prepended by the `name_shortcode` and a random string, since RADAR needs unique dataset titles in one workspace.

**Comment by jochenklar** [comment]
Refactor to use the RADAR credentials of the user, employing the proper OAuth2 workflow:

* The archive button in the react interface is a simple link of the form `/oauth/radar/archive?collection_id=<id>`. (I tested `window.open()`, but this will alert the popup protection in Firefox).
* From there the user will be redirected to RADAR to log in and then redirected back to  `/oauth/radar/callback` where the `code` is exchanged for the `access_token` which is stored in the session. (This is the OAuth2 part.)
* From there the user is redirected to `/oauth/radar/select` where they select the RADAR workspace (fetched from RADAR using the `access_token`) to use. Then the collection metadata is `POST`ed to RADAR, and if no errors occur, a job is started to (a) produce the file and (b) upload it to RADAR.
* The user is redirected to `/oauth/radar/export` which polls itself in a very simple way. When the job is done, the `export` action will "find" a `datasetUrl` in the metadata of the collection and will redirect the user to the RADAR page.

**Comment by jochenklar** [comment]
Rebased on `main` and force-pushed.

**Comment by jochenklar** [comment]
Ok, I fixed a few things and updated the RADAR schema to 9.1. I also tested it with RADAR and it works. PR can be reviewed now.

File: app/models/metadata.rb [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        'subjectArea' => self.metadata['subjectAreas']
+      }
+    end
+
+    if self.metadata['keywords']
+      radar_metadata['descriptiveMetadata']['keywords'] = {
+        'keyword' => self.metadata['keywords'].map {|keyword| {'value': keyword}}
+        end
+      }
+    end
```
Comment (phuang26) [inline app/models/metadata.rb:93]: please remove this line

File: app/models/metadata.rb [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    if self.metadata['subjectAreas']
+      radar_metadata['descriptiveMetadata']['subjectAreas'] = {
+        'subjectArea' => self.metadata['subjectAreas']
+      }
+    end
+
+    if self.metadata['keywords']
+      radar_metadata['descriptiveMetadata']['keywords'] = {
+        'keyword' => self.metadata['keywords'].map {|keyword| {'value': keyword}}
+        end
```
Comment (phuang26) [inline app/models/metadata.rb:91]: please remove this line

**Review by PiTrem — APPROVED** [review]

**Pull request merged**